legend-depot-store-mongo/src/main/java/org/finos/legend/depot/store/mongo/entities/EntitiesMongo.java

```diff
@@ -338,38 +344,55 @@ public List<Entity> getEntitiesByPackage(String groupId, String artifactId, Stri
         return entities.collect(Collectors.toList());
     }
 
-
     protected List<StoredEntity> transform(boolean summary, FindIterable query)
     {
         if (!summary)
         {
             return convert(query);
         }
         List<StoredEntity> result = new ArrayList<>();
-        query.forEach((Consumer<Document>)doc ->
+        query.forEach((Consumer<Document>) doc ->
         {
-            Map<String, Object> entity = (Map<String, Object>)doc.get(ENTITY);
-            result.add(new StoredEntityOverview(doc.getString(GROUP_ID), doc.getString(ARTIFACT_ID), doc.getString(VERSION_ID), doc.getBoolean(VERSIONED_ENTITY),(String)entity.get(PATH), (String)entity.get(CLASSIFIER_PATH)));
+            Map<String, Object> entity = (Map<String, Object>) doc.get(ENTITY);
+            result.add(new StoredEntityOverview(doc.getString(GROUP_ID), doc.getString(ARTIFACT_ID), doc.getString(VERSION_ID), doc.getBoolean(VERSIONED_ENTITY), (String) entity.get(PATH), (String) entity.get(CLASSIFIER_PATH)));
         });
         return result;
     }
 
     @Override
-    public List<StoredEntity> findReleasedEntitiesByClassifier(String classifier, boolean summary, boolean versionedEntities)
+    public List<StoredEntity> findReleasedEntitiesByClassifier(String classifier, String search, List<ProjectVersion> projectVersions, Integer limit, boolean summary, boolean versioned)
     {
-        return transform(summary, executeFind(and(and(eq(ENTITY_CLASSIFIER_PATH, classifier), eq(VERSIONED_ENTITY, versionedEntities)), ne(VERSION_ID, MASTER_SNAPSHOT))));
+        List<Bson> filters = new ArrayList<>();
+        filters.add(eq(ENTITY_CLASSIFIER_PATH, classifier));
+        filters.add(eq(VERSIONED_ENTITY, versioned));
+        if (projectVersions != null && !projectVersions.isEmpty())
+        {
+            filters.add(or(ListIterate.collect(projectVersions, projectVersion -> getArtifactAndVersionFilter(projectVersion.getGroupId(), projectVersion.getArtifactId(), projectVersion.getVersionId()))));
+        }
+        if (search != null)
+        {
+            filters.add(Filters.regex(ENTITY_PATH, Pattern.quote(search), "i"));
+        }
+        return transform(summary, executeFind(and(filters)).limit(Math.min(MAX_NUMBER_ENTITIES, limit == null ? Integer.MAX_VALUE : limit)));
```

Comment:
The logic to compute limit when the user supplied limit is null is already duplicated in a couple of places.

Can you lift that logic and pass the computed value ?

Reply:
only 2 places for now, I think I'll leave them separate unless really necessary to abstract